lib/Dialect/TTNN/IR/TTNNOps.cpp

```diff
@@ -121,6 +121,42 @@ ::mlir::LogicalResult mlir::tt::ttnn::MatmulOp::verify() {
 }
 // ANCHOR_END: adding_an_op_matmul_ttnn_verify
 
+::mlir::LogicalResult mlir::tt::ttnn::Conv2dOp::verify() {
+  ::mlir::RankedTensorType inputType = getInput().getType();
+  ::mlir::RankedTensorType weightType = getWeight().getType();
+  ::mlir::RankedTensorType biasType = getBias().getType();
+  // ::mlir::RankedTensorType outputType = getOutput().getType();
+  auto inputShape = inputType.getShape();
+  auto weightShape = weightType.getShape();
+  auto biasShape = biasType.getShape();
+  // auto outputShape = outputType.getShape();
+  if (inputShape.size() < 3) {
+    return emitOpError("Input must be at least a 3D tensor");
+  }
+  if (weightShape.size() != 4) {
+    return emitOpError("Weight must be a 4D tensor");
+  }
+  if (biasShape.size() != 4) {
+    return emitOpError("Weight must be a 4D tensor");
```

Comment:
"Weight" -> "Bias" in error msg.

Shouldn't the bias be a 1d tensor?

Reply:
Added that check